js/private/js_library.bzl

```diff
@@ -207,20 +207,23 @@ def _js_library_impl(ctx):
     sources = depset(transitive = [sources, additional_sources])
     types = depset(transitive = [types, additional_sources, additional_types])
 
-    transitive_sources = gather_transitive_sources(
-        sources = sources,
-        targets = ctx.attr.srcs + ctx.attr.types + ctx.attr.deps,
-    )
-
-    transitive_types = gather_transitive_types(
-        types = types,
```

Comment:
I think this (and the `gather_transitive_sources` above) are a unique case where the first param is a depset instead of list of files.

Is this because `js_library` is unique where it accepts `JsInfo`-producing targets as `srcs` + `types`?

This is the reason removing the use of these helpers in this unique spot is required for this PR...